internal/handler/userHandler.go

```diff
@@ -54,26 +54,37 @@ func (h *Handler) Save(context *gin.Context) {
 	}
 	user, err := mapper.UserRequestToUser(requestBody)
 	if err != nil {
-		logger.Error("Error while creating user. ", err)
+		logger.Error("Error while mapping request to user. ", err)
 		errorResponse := response.ErrorDto{
-			Message:   "Error while mapping request to user: " + err.Error(),
+			Message:   "Error while mapping request to user",
 			Timestamp: time.Now(),
 		}
 		context.IndentedJSON(http.StatusUnprocessableEntity, errorResponse)
 		return
 	}
 	createdUser, err := h.service.SaveUser(context, user)
+
 	if err != nil {
+		logger.Errorf("Error while connecting to database. %v", err)
```

Comment:
THis error message should be more generic. We don't know for sure it's a database issue.

Reply:
Fixed